compress.hh

```diff
@@ -18,9 +18,14 @@
 #include "seastarx.hh"
 
 class compressor {
-    sstring _name;
 public:
-    compressor(sstring);
+    enum class algorithm {
+        lz4,
+        zstd,
+        snappy,
+        deflate,
+        none,
```

Comment:
Do we in fact need none? We won't instantiate such a compressor, will we?

Reply:
`none` is what's in the schema when no compression is set.
We don't use `optional<compression_parameters>` anywhere, we have `compression_parameters` with `algorithm::none` instead.